src/client.rs

```diff
@@ -183,8 +183,27 @@ impl InstrumentedClientRequest {
     }
 }
 
+// convert http status code to span status following the rules described by the spec:
+// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#status
+fn convert_status(status: http::StatusCode) -> (StatusCode, String) {
+    match status.as_u16() {
+        100..=399 => {
+            (StatusCode::Unset, String::from(""))
```

Comment:
```suggestion
            (StatusCode::Unset, String::new())
```

Reply:
Actually these would probably be cleaner as `(StatusCode::Unset, None)` and have the caller do `msg.unwrap_or_default()`